tree/ntuple/v7/inc/ROOT/RNTupleProcessor.hxx

```diff
@@ -263,6 +298,65 @@ private:
    RNTupleChainProcessor(const std::vector<RNTupleOpenSpec> &ntuples, std::unique_ptr<RNTupleModel> model = nullptr);
 };
 
+// clang-format off
+/**
+\class ROOT::Experimental::RNTupleJoinProcessor
+\ingroup NTuple
+\brief Processor specializiation for horizontally concatenated RNTuples (joins).
+*/
+// clang-format on
+class RNTupleJoinProcessor : public RNTupleProcessor {
+   friend class RNTupleProcessor;
+
+private:
+   std::unique_ptr<RNTupleModel> fJoinModel;
+   /// Mapping of the auxiliary RNTuple name to its page source
+   std::unordered_map<std::string, std::unique_ptr<Internal::RPageSource>> fAuxiliaryPageSources;
+
+   bool fUseIndex = false;
+   std::vector<std::string> fJoinFieldNames;
+   /// Mapping of the auxiliary RNTuple name to its join index
+   std::unordered_map<std::string, std::unique_ptr<Internal::RNTupleIndex>> fJoinIndices;
```

Comment:
Instead of associative maps, would it be possible to establish a global "order" of ntuple names and have `fAuxiliaryPageSources` and `fJoinIndices` be a vector of items with the same order?